nuxeo-services/nuxeo-platform-web-common/src/main/resources/web/nuxeo.war/login.jsp

```diff
@@ -524,6 +524,16 @@ input:-webkit-autofill:focus {
 </div>
 
 <script type="text/javascript">
+  // Handle requested URL for Web UI
+  var webUIPrefix = 'ui'
+  var requestedURL = document.getElementById('requestedUrl').value;
+  if (!requestedURL || requestedURL.startsWith(webUIPrefix)) {
+    var indexOfHash = window.location.href.indexOf('#!');
+    if (indexOfHash > -1) {
+      document.getElementById('requestedUrl').value = webUIPrefix + '/' + window.location.href.substring(indexOfHash);
+    }
+  }
+
```

Comment:
Could we just ignore the Web UI prefix and always add the missing part if it contains `#!`? So that it will work for other webapp inside `nuxeo.war` such as `timeoff`.